src/components/analyses/listing/Listing.js

```diff
@@ -270,6 +290,42 @@ function Listing(props) {
         }
     );
 
+    const { isFetching: isFetchingTimeLimit } = useQuery({
+        queryKey: [VICE_TIME_LIMIT_QUERY_KEY, selected[0]],
+        queryFn: getTimeLimitForVICEAnalysis,
+        config: {
+            enabled: timeLimitQueryEnabled,
+            onSuccess: (resp) => {
+                setTimeLimit({
+                    [selected[0]]: formatDate(resp?.time_limit * 1000),
+                });
```

Comment:
So there are few reason why I thought of doing this:

 - There is a small gap between when the user click on the `Extend` button / menu item and the time loading mask appears because the `getTimeLimitForVICEAnalysis` query is controlled by `timeLimitQueryEnabled` state. So the user has a chance to click on the same button again or choose to extend time limit on another analysis or even do something else. To avoid potential race condition, I am storing the selected analysis id for which the timestamp is fetched. This should also help in debugging if the race condition still happens.
 - I thought of updating the estimated `Completion Time` for the selected analysis using this object but then @johnworth mentioned that can easily added to analyses listing itself. But I think we can still update estimated `Completion Time` when the time limit is extended by the user using this object. 

I can add some comments around the code.